test/infrastructure/docker/docker/kind_manager.go

```diff
@@ -50,23 +56,36 @@ func (m *Manager) CreateControlPlaneNode(name, image, clusterLabel, listenAddres
 		HostPort:      port,
 		ContainerPort: KubeadmContainerPort,
 	})
-	node, err := createNode(
-		name, image, clusterLabel, constants.ControlPlaneNodeRoleValue, mounts, portMappingsWithAPIServer,
-		// publish selected port for the API server
-		append([]string{"--expose", fmt.Sprintf("%d", port)}, labelsAsArgs(labels)...)...,
-	)
+	createOpts := &nodeCreateOpts{
+		Name:         name,
+		Image:        image,
+		ClusterName:  clusterName,
+		Role:         constants.ControlPlaneNodeRoleValue,
+		Port:         port,
+		PortMappings: portMappingsWithAPIServer,
+		Mounts:       mounts,
+	}
+	node, err := createNode(context.Background(), createOpts)
```

Comment:
I think we should pass through the context from the caller here (from the usage I could find, it shouldn't break anything)

Same for `CreateWorkerNode`

Reply:
+1

Reply:
Just to clarify, you I should change the `CreateControlPlaneNode` signature and plumb the context all the way through, right?

Reply:
Yup